src/ripple/peerfinder/impl/PeerfinderConfig.cpp

```diff
@@ -78,5 +87,53 @@ Config::onWrite(beast::PropertyStream::Map& map)
     map["ip_limit"] = ipLimit;
 }
 
+Config
+Config::makeConfig(
+    ripple::Config const& cfg,
+    std::uint16_t port,
+    bool validationPublicKey,
+    int ipLimit)
+{
+    PeerFinder::Config config;
+
+    config.legacyConfig = cfg.legacyPeersMax_;
```

Comment:
I looked at the code, it makes sense. One thing that caught my attention is how `ipLimit` is calculated. In the code prior to this PR, `inPeers` and `outPeers` are calculated without taking into account `wantIncoming` (I kept this logic). Consequently, `inPeers` is used in calculating `ipLimit`. With your approach, if `wantIncoming` is false then `inPeers` is 0 and `ipLimit` will be calculated differently. I think your approach is correct since we are not going to have any incoming connections when `wantIncoming` is false, and the original code is wrong. But I don't know what the original intention was.

Reply:
Thanks! Yes, I noticed that change, too, and forgot to point it out in my comment, but yeah, I did that. I had to change one of the unit tests to work with the new behavior, too. I'm fine with that, for the exact reason you specified. 

Looking through the git history, it looks like https://github.com/ripple/rippled/commit/0c67364e6c8694289eadeb561366cf97018309ca from 5 years ago is the relevant commit - it's where the limit was added, and it hasn't changed since then. Since the limit is explicitly for incoming connections, I don't think any real consideration was given to the value when no incoming connections are allowed, so I don't see a problem with changing it.

Reply:
Ok, so I'll keep this new behavior. I did find this because the unit test failed.